cpp/include/cudf/strings/convert/convert_fixed_point.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace cudf {
+namespace strings {
+/**
+ * @addtogroup strings_convert
+ * @{
+ * @file
+ */
+
+/**
+ * @brief Returns a new fixed-point column parsing decimal values from the
+ * provided strings column.
+ *
+ * Any null entries will result in corresponding null entries in the output column.
+ *
+ * Only characters [0-9] plus a prefix '-' and '+' and a single decimal point
+ * are recognized. When any other character is encountered, the parsing ends
+ * for that string and the current digits are converted into a fixed-point value.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ['123', '-876', '543.2', '-0.12']
+ * dt = {DECIMAL32, scale=-2}
+ * fp = to_fixed_point(s, dt)
+ * fp is [123400, -87600, 54320, -12]
+ * @endcode
+ *
+ * Overflow of the resulting value type is not checked.
+ * The scale in the `output_type` is used for setting the integer component.
+ *
+ * @throw cudf::logic_error if `output_type` is not a fixed-point decimal type.
+ *
+ * @param strings Strings instance for this operation.
+ * @param output_type Type of fixed-point column to return including the scale value.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column of `output_type`.
+ */
+std::unique_ptr<column> to_fixed_point(
```

Comment:
Seems like this could more generically be called `column_from_strings` or something like that, and the fact that fixed-point output is desired would be inferred from the `output_type` parameter. The same function could then support floating point, boolean, datetime, etc.

Reply:
The other convert functions also accept appropriate output types (i.e. integer, floating, duration, timestamp). We could possible create an uber `column_from_strings` that would call each of the detail APIs appropriately but that would seem out of scope for this PR.
https://docs.rapids.ai/api/libcudf/nightly/group__strings__convert.html